GW2EIEvtcParser/EncounterLogic/Raids/W5/StatueOfIce.cs

```diff
@@ -107,14 +109,63 @@ internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog lo
                 default:
                     break;
             }
+        }
+
+        internal override void ComputeEnvironmentCombatReplayDecorations(ParsedEvtcLog log)
+        {
+            base.ComputeEnvironmentCombatReplayDecorations(log);
+
+            // Numbing Breach - Cracks - White smoke indicator
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.BrokenKingNumbingBreachIndicator, out IReadOnlyList<EffectEvent> cracksIndicators))
+            {
+                foreach (EffectEvent effect in cracksIndicators)
+                {
+                    (long, long) lifespan = effect.ComputeLifespan(log, 1000);
+                    var connector = new PositionConnector(effect.Position);
+                    var circle = new CircleDecoration(115, lifespan, "rgba(219, 233, 244, 0.2)", connector);
+                    EnvironmentDecorations.Add(circle);
+                }
+            }
+
+            // Numbing Breach - Cracks - Damage zone
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.BrokenKingNumbingBreachDamage, out IReadOnlyList<EffectEvent> cracks))
+            {
+                foreach (EffectEvent effect in cracks)
+                {
+                    (long, long) lifespan = effect.ComputeLifespan(log, 30000);
+                    var connector = new PositionConnector(effect.Position);
+                    var rotationConnector = new AngleConnector(effect.Rotation.Z);
+                    var rectangle = (RectangleDecoration)new RectangleDecoration(40, 230, lifespan, "rgba(66, 130, 253, 0.2)", connector).UsingRotationConnector(rotationConnector);
+                    EnvironmentDecorations.Add(rectangle);
+                }
+            }
 
+            // Hailstorm - Greens
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.BrokenKingHailstormGreen, out IReadOnlyList<EffectEvent> greens))
+            {
+                foreach (EffectEvent green in greens)
+                {
+                    // Ice Breaker - Failed Greens
+                    log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.BrokenKingIceBreakerGreenExplosion, out IReadOnlyList<EffectEvent> failedGreens);
+                    EffectEvent failedGreen = failedGreens.FirstOrDefault(x => x.Position.X == green.Position.X && x.Position.Y == green.Position.Y);
```

Comment:
x.Position.Distance2DToPoint(green.Position) < 1e-6
you should also do a Abs(x.Time - green.Time - 15000) < some threshold, for the rare situations where 2 failed greens may have happened on exactly the same spot

Reply:
That code will throw a null exception if BrokenKingIceBreakerGreenExplosion is not present in the log
failedGreen = log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.BrokenKingIceBreakerGreenExplosion, out IReadOnlyList<EffectEvent> failedGreens) ? ... : null